processor/metricextractprocessor/config.go

```diff
@@ -50,31 +54,58 @@ const (
 )
 
 var (
-	// errExtractMissing is the error message for a missing extract expression.
-	errExtractMissing = errors.New("extract expression is required")
+	// errExprExtractMissing is the error message for a missing expr extract expression.
+	errExprExtractMissing = errors.New("extract expression is required")
+
+	// errExtractMissing is the error message for a missing ottl extract expression.
+	errOTTLExtractMissing = errors.New("ottl_extract expression is required")
 
 	// errMetricTypeInvalid is the error message for an invalid metric type.
 	errMetricTypeInvalid = errors.New("invalid metric type")
 )
 
 // Config is the config of the processor.
 type Config struct {
-	Route      string            `mapstructure:"route"`
-	Match      string            `mapstructure:"match"`
-	Extract    string            `mapstructure:"extract"`
-	MetricName string            `mapstructure:"metric_name"`
-	MetricUnit string            `mapstructure:"metric_unit"`
-	MetricType string            `mapstructure:"metric_type"`
-	Attributes map[string]string `mapstructure:"attributes"`
+	Route          string            `mapstructure:"route"`
+	Match          *string           `mapstructure:"match"`
+	OTTLMatch      *string           `mapstructure:"ottl_match"`
+	Extract        string            `mapstructure:"extract"`
+	OTTLExtract    string            `mapstructure:"ottl_extract"`
+	MetricName     string            `mapstructure:"metric_name"`
+	MetricUnit     string            `mapstructure:"metric_unit"`
+	MetricType     string            `mapstructure:"metric_type"`
+	Attributes     map[string]string `mapstructure:"attributes"`
+	OTTLAttributes map[string]string `mapstructure:"ottl_attributes"`
 }
 
 // Validate validates the config.
 func (c Config) Validate() error {
+	usesExprFields := c.Extract != "" || c.Match != nil || c.Attributes != nil
+	usesOTTLFields := c.OTTLExtract != "" || c.OTTLMatch != nil || c.OTTLAttributes != nil
```

Comment:
nit: you already have IsOTTL below. might be easier to just use that?

Reply:
I think it might be confusing here why isOTTL is relevant. It's also not a direct substitute I don't think? Mostly I just feel it's clearer written this way, though.